code/server/server.cpp

```diff
@@ -0,0 +1,178 @@
+#include <iostream>
+#include <yarp/os/all.h>
+
+using namespace std;
+using namespace yarp::os;
+
+#define FIFO_SIZE 10
+
+volatile int intFIFOtmp[FIFO_SIZE];
+volatile int intFIFOnum;
+
+class FIFO{
+private:
+	int intHead;
+	int intQueLen;
+	int intFIFO[FIFO_SIZE];
+public:
+	FIFO();
+	int enqueue(int intEnqData);
+	int dequeue();
+	int head_value();
+	void delete_element(int intNatural);
+	void FIFO_copy();
+};
+
+FIFO::FIFO()	//constructor
+{
+	intHead = 0;
+	intQueLen = 0;
+	for(int i=0;i<FIFO_SIZE;i++){
+		intFIFO[i]=0;
+	}
+}
+
+int FIFO::enqueue(int intEnqData)	//enqueue the argument "intEnqData"
+{
+	if(intQueLen < FIFO_SIZE){
+		intFIFO[(intHead + intQueLen) % FIFO_SIZE] = intEnqData;
+		intQueLen++;
+		FIFO_copy();	//copy the content of FIFO to global variable for monitoring
+		return true;
+	}else{
+		return false;
+	}
+}
+
+int FIFO::dequeue()		//dequeue the first element of FIFO
+{
+	if(intQueLen>0){
+		intQueLen--;
+		if(intHead+1 < FIFO_SIZE){
+			intHead++;
+		}else{
+			intHead = 0;
+		}
+		FIFO_copy();	//copy the content of FIFO to global variable for monitoring
+		return true;
+	}else{
+		return false;
+	}
+}
+
+int FIFO::head_value()	//return the value of the first element of FIFO
+{
+	return (intFIFO[intHead]);
+}
+
+void FIFO::delete_element(int intNatural)	//delete the element of FIFO corespoinding to the argument "intNatural"
+{
+	int intLen = intQueLen;
+	for(int i=0; i<intLen; i++){
+		if(intNatural == intFIFO[intHead]){
+			dequeue();
+		}
+		else{
+			enqueue(intFIFO[intHead]);
+			dequeue();
+		}
+	}
+}
+
```

Comment:
I'd need to get a better insight of what's going on here to provide hints that are spot on.